File: tests/ws_server/ws_test_server.py
import asyncio
import websockets
from datetime import datetime

async def handle(ws, path=None):
    print(f"[CONNECT] {datetime.now().isoformat()}")
    try:
        async for message in ws:
            pass  # 接收上行音频
    except websockets.exceptions.ConnectionClosed:
        pass
    finally:
        print(f"[DISCONNECT] {datetime.now().isoformat()}")

async def main():
    async with websockets.serve(handle, "0.0.0.0", 8080):
        print("WebSocket test server listening on port 8080")
        await asyncio.Future()  # run forever

asyncio.run(main())

File: tests/ws_server/ws_test_server.js
const WebSocket = require('ws');
const server = new WebSocket.Server({ port: 8080 });

let connCount = 0;

server.on('connection', (ws, req) => {
  connCount++;
  const id = connCount;
  console.log(`[CONNECT] ${new Date().toISOString()} #${id} from ${req.socket.remoteAddress} (total: ${server.clients.size})`);

  ws.on('message', (data) => {
    // 上行音频接收 - 仅统计
  });

  ws.on('close', () => {
    console.log(`[DISCONNECT] ${new Date().toISOString()} #${id} (total: ${server.clients.size})`);
  });

  ws.on('error', (err) => {
    console.error(`[ERROR] #${id} ${err.message}`);
  });
});

console.log('WebSocket test server listening on port 8080');
